include/pybind11/detail/common.h

```diff
@@ -459,7 +459,16 @@ enum class return_value_policy : uint8_t {
         collected while Python is still using the child. More advanced
         variations of this scheme are also possible using combinations of
         return_value_policy::reference and the keep_alive call policy */
-    reference_internal
+    reference_internal,
+
+    /** With this policy, C++ string types are converted to Python bytes,
+        instead of str. This is most useful when a C++ function returns a
+        container-like type with nested C++ string types, and py::bytes cannot
+        be applied easily. Note that this return_value_policy is not concerned
+        with lifetime/ownership semantics, like the other policies, but the
+        purpose of return_as_bytes is certain to be orthogonal, because C++
+        strings are always copied to Python bytes or str. */
```

Comment:
Can we mark this as experimental and likely to change in the future?

Reply:
@wangxf123456 Please add this: https://github.com/pybind/pybind11/pull/3838#issuecomment-1085141637 here too